adapters/adkernelAdn/adkernelAdn.go

```diff
@@ -0,0 +1,254 @@
+package adkernelAdn
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/golang/glog"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const defaultDomain string = "tag.adkernel.com"
+
+type adkernelAdnAdapter struct{}
+
+//MakeRequests prepares request information for prebid-server core
+func (adapter *adkernelAdnAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+	if len(request.Imp) == 0 {
+		errs = append(errs, newBadInputError("No impression in the bid request"))
+		return nil, errs
+	}
+	imps, impExts, err := getImpressionsInfo(&request.Imp)
+	if len(*imps) == 0 {
+		return nil, *err
+	}
+	errs = append(errs, *err...)
+
+	pub2impressions, dispErrors := dispatchImpressions(imps, impExts)
+	if len(*dispErrors) > 0 {
+		errs = append(errs, *dispErrors...)
+	}
+	if len(pub2impressions) == 0 {
+		return nil, errs
+	}
+	result := make([]*adapters.RequestData, 0, len(pub2impressions))
+	for k, imps := range pub2impressions {
+		bidRequest, err := buildAdapterRequest(request, &k, &imps)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			result = append(result, bidRequest)
+		}
+	}
+	return result, errs
+}
+
+// getImpressionsInfo checks each impression for validity and returns impressions copy with corresponding exts
+func getImpressionsInfo(imps *[]openrtb.Imp) (*[]openrtb.Imp, *[]openrtb_ext.ExtImpAdkernelAdn, *[]error) {
```

Comment:
The code throughout can be simplified if you drop the `*`s in the method signatures. A pointer to a slice doesn't make much sense as a type, because slices [are already pointers](https://blog.golang.org/go-slices-usage-and-internals#TOC_4.).

Reply:
Fixed in e341d28059d36a1fb21dc9a01d4479cac76c2059